lib/modules/apostrophe-attachments/lib/api.js

```diff
@@ -874,70 +877,97 @@ module.exports = function(self, options) {
     var commands = [];
 
     if (!doc.trash) {
+      if (ids.length > 0) {
```

Comment:
[Code proposal]
To avoid code redundancy you can do something like that:
(it replaces all the if else blocks of this function)
```javascript
    const commands = [
      ...ids.length ? [
        [
          {
            _id: { $in: ids }
          },
          {
            $addToSet: { [!doc.trash ? 'docIds' : 'trashDocIds']: doc._id }
          }
        ],
        [
          {
            _id: { $in: ids }
          },
          {
            $pull: { [!doc.trash ? 'trashDocIds' : 'docIds']: doc._id }
          }
        ],
        [
          {
            $or: [
              {
                trashDocIds: { $in: [ doc._id ] }
              },
              {
                docIds: { $in: [ doc._id ] }
              }
            ],
            _id: { $nin: ids }
          },
          {
            $pull: {
              trashDocIds: doc._id,
              docIds: doc._id
            }
          }
        ],
        [
          {
            _id: { $in: ids }
          },
          {
            $set: {
              utilized: true
            }
          }
        ]
      ] : [
        // This is the only command that must be run when
        // the document now has no attachments
        [
          {
            $or: [
              {
                trashDocIds: { $in: [ doc._id ] }
              },
              {
                docIds: { $in: [ doc._id ] }
              }
            ]
          },
          {
            $pull: {
              trashDocIds: doc._id,
              docIds: doc._id
            }
          }
        ]
      ]
    ];
```